### ISSUE TITLE
Idea: 2 stage bundle pipeline

### ISSUE THREAD

**Issue by jeremy-coleman** [fuse-box/fuse-box#1572]
This is just an idea.
Tldr: separate the transpile and bundle process.
I started using this strategy with nextjs and browserify projects, and found it results in siginificantly less issues related to bundling. I think the v4 of fuse box could make use of it also. Basically, pre transpile everything , optionally write it to disk, and only introduce files to the bundler that are finalized for browser use. This also allows the user infinite flexibility. For ex: imagine a typescript react project with tsx, some scss files, maybe some gql and json and some svg files. Now, just run tsc , node sass, and maybe some postcss cli tasks using their native implementations. The output should be directed to ./fusebox/staging. This stages dev experience would be much like a gulp task and should be handled by sparky. This would allow fusebox to really only be concerned with js ,css , and html and efficiently bundling them. Also the typescript watcher in the 3.5 release is crazy fast WITH type checks when using the incremental option ( ~<1s on 100k lines). Of course fuse box could still provide its own transpiler , but it wouldnt be required

**Comment by nchanged** [comment]
@jeremy-coleman it's already been implemented ;-) 
Check [here](https://github.com/fuse-box/fuse-box/blob/v4/docs/cache.md#ftl-mode)

**Comment by nchanged** [comment]
closing this for now
